src/test/java/DataFetcherWithErrorsTest.java

```diff
@@ -0,0 +1,99 @@
+import static org.junit.Assert.assertTrue;
+
+
+public class DataFetcherWithErrorsTest {
+
+    @Test
+    public void testDataFetcherWithErrors() {
+        GraphQLError error = new GraphQLError() {
+
+            @Override
+            public String getMessage() {
+                return "badField is bad";
+            }
+
+            @Override
+            public List<SourceLocation> getLocations() {
+                return singletonList(new SourceLocation(2, 10));
+            }
+
+            @Override
+            public ErrorType getErrorType() {
+                return ErrorType.DataFetchingException;
+            }
+
+            @Override
+            public List<Object> getPath() {
+                return Arrays.asList("child", "badField");
+            }
+        };
+        DataFetcher dataFetcher = environment -> new DataFetcherResult<Map>(
+                singletonMap("child",
+                        new HashMap<String, Object>() {{
+                            put("goodField", "good");
+                            put("badField", null);
+                        }}
+                ),
+                singletonList(error)
+        );
+
+        String schema = "type Query{root: Root}\n" +
+                "type Root{ parent: Parent}\n" +
+                "type Parent{ child: Child}\n" +
+                "type Child{\n" +
+                "  goodField: String,\n" +
+                "  badField: String\n" +
+                "}";
+        SchemaParser schemaParser = new SchemaParser();
+        TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(schema);
+        RuntimeWiring runtimeWiring = newRuntimeWiring()
+                .type("Root", builder -> builder.dataFetcher("parent", dataFetcher))
+                .type("Query", builder -> builder.dataFetcher("root", new StaticDataFetcher(emptyMap())))
+                .build();
+
+        SchemaGenerator schemaGenerator = new SchemaGenerator();
+        GraphQLSchema graphQLSchema = schemaGenerator.makeExecutableSchema(typeDefinitionRegistry, runtimeWiring);
+
+        GraphQL build = GraphQL.newGraphQL(graphQLSchema).build();
+        ExecutionResult executionResult = build.execute("{root{parent{child{goodField, badField}}}}");
+        System.out.println(executionResult.getData().toString());
+
+        assertEquals(1, executionResult.getErrors().size());
+        GraphQLError firstError = executionResult.getErrors().get(0);
+        assertEquals(Arrays.asList("root", "parent", "child", "badField"), firstError.getPath());
+        assertEquals("badField is bad", firstError.getMessage());
+        assertEquals(singletonList(new SourceLocation(3, 17)), firstError.getLocations());
+
+        Map<String, String> child = ((Map<String, Map<String, Map<String, Map<String, String>>>>) executionResult.getData()).get("root").get("parent").get("child");
+        assertEquals("good", child.get("goodField"));
+        assertEquals(null, child.get("badField"));
+
+    }
```

Comment:
Can we get some tests around when the downstream error path is empty and so on and hence the current path is used as is.

Also should we detect if the path is exactly the same and use it instead.  eg

Imagine you are running on `[hero,friends]` and you get a DFResult with an error with `[hero,friends]` as the path. 

is it "[hero,friends,hero,friends]" or is it just the same path??

Reply:
The path should be relative to the data fetcher location, so it is up to the fetcher how it creates that path.